repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,227 @@
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+from os import listdir
+from os.path import isfile, join, isdir
+import re
+
+
+def get_config_files(rootfile):
+    """ return MariaDB config files """
+
+    if not isfile(rootfile):
+        return []
+
+    files = [rootfile]
+    dirs = []
+    def get_new_files(directory):
+        new_configs = []
+        for f in listdir(directory):
+            path = join(directory, f)
+            if isfile(path) and path.endswith('.cnf') and path not in files:
+                new_configs.append(path)
+        return new_configs
+
+    def get_new_dirs(configfile):
+        values = []
+        with open(configfile, "r") as f:
+            m = re.findall(r"^\s*!includedir[\s=]+(.+)$" ,f.read(), flags=re.MULTILINE)
+            if m:
+                for new_dir in m:
+                    new_dir = new_dir.strip()
+                    if isdir(new_dir) and new_dir not in dirs:
+                        values.append(new_dir)
+        return values
+
+    i = 0
+    while i < len(files):
+        configfile = files[i]
+        includedirs = get_new_dirs(configfile)
+        dirs = dirs + includedirs
+        for directory in includedirs:
+            files = files + get_new_files(directory)
+        i += 1
+
+    return files
+
+
+def get_configs_as_text(files):
+    """ concatenate files into big config text """
+
+    ret = ""
+    for path in files:
+        with open(path) as f:
+            ret = "\n".join([ret, f.read()])
+    return ret
+
+
+def get_used_opts(config):
+    """ parse config text into list """
+
+    lines = []
+    for line in config.splitlines():
+        line = line.strip()
+        if not line or line.startswith('#') or line.startswith(';') or line.startswith('['):
+            continue
+        lines.append(line)
+    return lines
+
+
+def get_plugin_dir(config):
+    """ return plugin dir """
+
+    m = re.findall(r"^\s*plugin[-_]dir[\s=]+(.+)$", config, flags=re.MULTILINE)
+    if m:
+        # TODO what if there are more in config
+        return m[0].strip()
+    else:
+        # RHEL7 default plugin dir
+        return "/usr/lib64/mysql/plugin"
+
+
+def get_unsupported_plugins(plugin_dir):
+    """ return list of unsupported plugins """
+
+    plugins_in8 = [
```

Comment:
I'd suggest to name this and other variables more specifically, if we ever want to show changes between different streams, then `el8` is gonna be ambiguous.. maybe something like `plugins_55`, `plugins_103`, ... Same below for the options.